integration-test/1252-roads-surface.py

```diff
@@ -0,0 +1,52 @@
+# Add surface properties to roads layer (at max zooms)
+
+# Prince St with cobblestones in Alexandria, VA
+# https://www.openstreetmap.org/way/190536019
+test.assert_has_feature(
+    15, 9371, 12546, 'roads',
+    { 'id': 190536019, 'kind': 'minor_road', 'surface': 'cobblestone'})
+
+# and that surface property stays at earlier zooms
+test.assert_has_feature(
+    13, 2342, 3136, 'roads',
+    { 'id': 190536019, 'kind': 'minor_road', 'surface': 'cobblestone'})
+
+
+# motorway in Kraków, Poland
+# http://www.openstreetmap.org/way/431783017
+test.assert_has_feature(
+    12, 2273, 1388, 'roads',
+    { 'id': 431783017, 'kind_detail': 'motorway', 'surface': 'asphalt'})
+
+
+# But strip that surface property off at earlier zooms
+test.assert_no_matching_feature(
+    7, 71, 43, 'roads',
+    { 'kind_detail': 'motorway', 'surface': 'asphalt'})
+
+# track with cycling route in Schartau, Germany
+# http://www.openstreetmap.org/way/58691615
+
+test.assert_has_feature(
+    15, 17456, 10780, 'roads',
+    { 'id': 58691615, 'kind_detail': 'track', 'surface': 'concrete:lanes'})
+
+test.assert_has_feature(
+    13, 4364, 2695, 'roads',
+    { 'id': 58691615, 'kind_detail': 'track', 'surface': 'concrete:lanes'})
+
+test.assert_has_feature(
+    11, 1091, 673, 'roads',
+    { 'id': 58691615, 'kind_detail': 'track', 'surface': 'concrete:lanes'})
+
+test.assert_has_feature(
+    10, 545, 336, 'roads',
+    { 'id': 58691615, 'kind_detail': 'track', 'surface': 'concrete:lanes'})
```

Comment:
Strangely enough **concrete:lanes** is a valid `surface` value in OSM:

- http://wiki.openstreetmap.org/wiki/Key:surface

But in Tilezen we generally avoid `:` and `-` in property names or values in favor of `_`.

That's controlled in the YAML config. So update where `surface` is added to include a mapping:

- https://github.com/tilezen/vector-datasource/blob/master/yaml/roads.yaml#L32

More like how we trap aerialway values to prevent `j-bar` from being exported in with the case statement:

https://github.com/tilezen/vector-datasource/blob/master/yaml/roads.yaml#L633

```
  - filter:
      aerialway:
        - drag_lift
        - platter
        - t-bar
        - goods
        - magic_carpet
        - rope_tow
        - "yes"
        - zip_line
        - j-bar
        - unknown
        - mixed_lift
        - canopy
        - cableway
    min_zoom: 15
    table: osm
    output:
      <<: *osm_highway_properties
      <<: *osm_network
      kind: aerialway
      kind_detail:
        case:
          - { when: { aerialway: 't-bar' }, then: 't_bar' }
          - { when: { aerialway: 'j-bar' }, then: 'j_bar' }
          - { when: { aerialway: 'yes' }, then: null }
          - else: { col: 'aerialway' }
```